collections/nemo_nlp/nemo_nlp/data/datasets/joint_intent_slot.py

```diff
@@ -39,37 +39,47 @@ def get_features(queries,
                  pad_label=128,
                  raw_slots=None):
     all_subtokens = []
-    all_slot_masks = []
+    all_loss_mask = []
+    all_subtokens_mask = []
     all_segment_ids = []
     all_input_ids = []
-    all_input_masks = []
+    all_input_mask = []
     sent_lengths = []
     all_slots = []
+
     with_label = False
     if raw_slots is not None:
         with_label = True
 
     for i, query in enumerate(queries):
         words = query.strip().split()
         subtokens = ['[CLS]']
-        slot_mask = [True]  # True if a token is the start of a new word
+        loss_mask = [True]  # True if a token is the start of a new word
+        subtokens_mask = [False]
         if with_label:
             slots = [pad_label]
 
         for j, word in enumerate(words):
             word_tokens = tokenizer.tokenize(word)
             subtokens.extend(word_tokens)
-            slot_mask.append(True)
-            slot_mask.extend([False] * (len(word_tokens) - 1))
+
+            loss_mask.append(True)
+            loss_mask.extend([True] * (len(word_tokens) - 1))
```

Comment:
should we add an option for users to choose whether to take into the loss of non-beginning tokens? If that option exists, then we should process loss_mask accordingly

Reply:
Added two parameters: 1-mask the extra subtokens, and 2-[CLS] and [SEP] tokens.